draft-bbs-signatures.md

```diff
@@ -439,205 +452,269 @@ Procedure:
+                 operation.
+- signature (REQUIRED), an octet string of the form outputted by the
+                        Sign operation.
+- header (OPTIONAL), an octet string containing application specific
+                     information. If not supplied, it defaults to an
+                     empty string.
 - ph (OPTIONAL), octet string.
-- msg_1,..., msg_L (OPTIONAL), octet strings. Messages in input to Sign.
-- H_1,..., H_L (OPTIONAL), points of G1. The generators in input to Sign.
-- RevealedIndexes (OPTIONAL), vector of unsigned integers. Indexes of revealed messages.
+- signed (OPTIONAL), a list of tuples containing a generator (point on
+                     G1), a non-zero scalar mod r representing a message
+                     (see above for requirements).
+- revealedIndexes (OPTIONAL), a list of distinct integers, in the range
+                              1 to length(signed), in ascending order
+                              (see above for requirements).
 
 Parameters:
 
-- Ciphersuite_ID (REQUIRED), ASCII string. The unique ID of the ciphersuite.
-- H_s (REQUIRED), point of G1. The generator for the blinding value of the signature.
-- H_d (REQUIRED), point of G1. The generator used to sign the signature domain.
+- Ciphersuite_ID (REQUIRED), ASCII string. The unique ciphersuite ID.
+- H_s (REQUIRED), point of G1. The generator for the blinding value of
+                  the signature.
+- H_d (REQUIRED), point of G1. The generator used to sign the signature
+                  domain.
 
 Definitions:
 
-- L, is the non-negative integer representing the number of messages to be signed e.g length(msg_1,...,msg_L). Note if no messages are supplied as an input to this operation, the value of L MUST evaluate to zero (0).
-- R, is the non-negative integer representing the number of revealed messages e.g length(RevealedIndexes). Note if no revealed messages are supplied as an input to this operation, the value of R MUST evaluate to zero (0).
+- L, is the non-negative integer representing the number of signed
+     messages i.e., L = length(signed). Note, if signed is not supplied
+     as an input, the value of L MUST evaluate to zero (0).
+- R, is the non-negative integer representing the number of revealed
+     messages i.e., R = length(revealedIndexes). If revealedIndexes is
+     not supplied as an input, the value of R MUST evaluate to zero (0).
+- U, is the non-negative integer representing the number of un-revealed
+     messages i.e., U = L - R.
 
 Outputs:
 
 - proof, octet string; or INVALID.
 
 Procedure:
 
-1. signature_result = octets_to_signature(signature)
+1. if length(revealedIndexes) > L, return INVALID
 
-2. (i1, i2,..., iR) = RevealedIndexes
+2. [i1, ..., iR] = revealedIndexes
 
-3. (j1, j2,..., jU) = [L] \ RevealedIndexes
+3. [j1, ..., jU] = (1, ..., L) \ revealedIndexes
 
-4. if signature_result is INVALID, return INVALID
+4. (H_i1, msg_i1), ..., (H_iR, msg_iR) = signed[i1], ..., signed[iR]
```

Comment:
Currently unused, but I guess it will be when the revealed messages and indexes are hashed